src/lib/parts.ts

```diff
@@ -60,22 +61,32 @@ export class AttributeCommitter {
   }
 
   /**
-   * Creates a single part. Override this to create a differnt type of part.
+   * Creates a single part. Override this to create a different type of part.
    */
   protected _createPart(): AttributePart {
     return new AttributePart(this);
   }
 
-  protected _getValue(): unknown {
+  getValue(): unknown {
     const strings = this.strings;
     const l = strings.length - 1;
     let text = '';
+    let changed = false;
 
     for (let i = 0; i < l; i++) {
       text += strings[i];
       const part = this.parts[i];
       if (part !== undefined) {
         const v = part.value;
+        if (v !== noChange) {
```

Comment:
Yes, basically. On the client, it's `commit` which notices there's a `noChange` value from a directive and avoids calling `commit` on the committer. For SSR, we don't want to call commit, but we do want to get the concatenated value using this function (since there is non-trivial logic in it). So this threads `noChange` through to say "don't emit this".

As noted in the comment below, this doesn't actually work for multiple attribute bindings, but it's broken on the client also (if one part commits and the other was `noChange`, it gets `'[object Object]'` concatenated in.  There's already an issue open for that: https://github.com/Polymer/lit-html/issues/1066

https://github.com/Polymer/lit-html/issues/1066